client/signup/steps/woocommerce-install/confirm/index.tsx

```diff
@@ -1,133 +1,139 @@
+	margin-bottom: 40px;
+`;
+
+function SupportLink() {
+	return (
+		<p>
+			{ createInterpolateElement( __( 'Need help? <a>Contact support</a>' ), {
+				a: <SupportLinkStyle href="#support-link" />,
+			} ) }
+		</p>
+	);
+}
+
 export default function Confirm( props: WooCommerceInstallProps ): ReactElement | null {
-	const { goToStep, isReskinned, stepSectionName, headerTitle, headerDescription } = props;
+	const { goToStep, isReskinned, headerTitle, headerDescription } = props;
 	const { __ } = useI18n();
 
 	// selectedSiteId is set by the controller whenever site is provided as a query param.
 	const siteId = useSelector( getSelectedSiteId ) as number;
 
 	const {
-		isFetching,
 		wpcomDomain,
 		stagingDomain,
-		eligibilityHolds,
-		eligibilityWarnings,
 		wpcomSubdomainWarning,
 		siteUpgrading,
+		hasBlockers,
+		warnings,
 	} = useWooCommerceOnPlansEligibility( siteId );
 
-	const isLoading = ! siteId || isFetching;
-
-	const backUrl =
-		stepSectionName === 'info' || typeof stepSectionName === 'undefined'
-			? `/woocommerce-installation/${ wpcomDomain }`
-			: 'info';
-
 	function getWPComSubdomainWarningContent() {
-		return (
-			<>
-				<div className="confirm__info-section">{ isLoading && <LoadingEllipsis /> }</div>
-
-				<div className="confirm__instructions-container">
-					<div className="confirm__instructions-title confirm__instructions-wpcom-domain">
-						{ wpcomDomain }
-					</div>
-
-					<div className="confirm__instructions-title">{ stagingDomain }</div>
+		if ( ! wpcomSubdomainWarning ) {
+			return null;
+		}
 
-					<p>
-						{ __(
-							'By installing this product your subdomain will change. You can change it later to a custom domain and we will pick up the tab for a year.'
-						) }
-					</p>
-
-					<p>
-						{ createInterpolateElement( __( '<a>Contact support</a> for help and questions.' ), {
-							a: <a href="#support-link" />,
-						} ) }
-					</p>
-
-					<NextButton
-						disabled={ isLoading }
-						onClick={ () => {
-							return goToStep( 'confirm', 'plan' );
-						} }
-					>
-						{ __( 'Sounds good' ) }
-					</NextButton>
-				</div>
-			</>
-		);
+		return <DomainEligibilityWarning wpcomDomain={ wpcomDomain } stagingDomain={ stagingDomain } />;
 	}
 
 	function getCheckoutContent() {
-		return (
-			<>
-				<div className="confirm__info-section">{ isLoading && <LoadingEllipsis /> }</div>
+		if ( ! siteUpgrading.required ) {
+			return null;
+		}
 
-				<div className="confirm__instructions-container">
-					<div className="confirm__instructions-title">{ __( 'Upgrading is required' ) }</div>
-
-					<p>{ siteUpgrading.description }</p>
-
-					<NextButton
-						disabled={ isLoading }
-						onClick={ () => {
-							window.location.href = siteUpgrading.checkoutUrl;
-						} }
-					>
-						{ siteUpgrading.checkoutText }
-					</NextButton>
-				</div>
-			</>
+		return (
+			<PlanWarning title={ __( 'Upgrade your plan' ) }>{ siteUpgrading.description }</PlanWarning>
```

Comment:
Do we need to translate `siteUpgrading.description`? Or is the server responsible for this translation?

Reply:
it's [already translated](https://github.com/Automattic/wp-calypso/blob/c79cff3f5b77a6c951581a94ef438c9452c6cd84/client/signup/steps/woocommerce-install/hooks/use-woop-handling/index.ts#L127-L133) in the hook.